docs/external-database.asciidoc

```diff
@@ -0,0 +1,50 @@
+= External database support
+
+Keycloak Operator allows you to use an external Postgresql database by modifying `keycloak-postgresql` Endpoints object.
+The Operator also reuses the same `keycloak-db-secret` Secret but will require two additional properties:
+
+* POSTGRES_EXTERNAL_ADDRESS - an IP address (or a Hostname) or the external database. This address needs
+  to be resolvable from Kubernetes cluster perspective.
+* POSTGRES_EXTERNAL_PORT - (Optional) A database port.
+
+The other properties remain the same for both operating modes (hosted Postgresql and an external one).
+
```

Comment:
@slaskawi Thank you for adding docs on this. It will be very useful to the end users of this operator.

I think this section should just first list what the user needs to provide to allow external database support and then if we still want to go into an explanation on how it is done internally in the operator.

```suggestion
By default, the operator will create a Postgres instance and database secret with each Keycloak CR. To enable external database support, you need to do two steps.

1. Create a Secret resource with the fields listed below
2. Specify the spec.externalDatabase.databaseSecretName in the Keycloak CR spec

 *  POSTGRES_DATABASE: //TODO
 *  POSTGRES_HOST:  //TODO
 *  POSTGRES_PASSWORD:  //TODO
*   POSTGRES_SUPERUSER:  //TODO
*   POSTGRES_USERNAME:  //TODO
* POSTGRES_EXTERNAL_ADDRESS - an IP address (or a Hostname) or the external database. This address needs
  to be resolvable from Kubernetes cluster perspective.
* POSTGRES_EXTERNAL_PORT - (Optional) A database port.

> Here's an example of such secret (note that values are Base64 encoded, see https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets[Kubernetes Secrets manual]:
> 
> .`keycloak-db-secret` Secret
> ```yaml
> apiVersion: v1
> data:
>   POSTGRES_DATABASE: cm9vdA==
>   POSTGRES_EXTERNAL_ADDRESS: MTcyLjE3LjAuMw==
>   POSTGRES_EXTERNAL_PORT: NTQzMg==
>   POSTGRES_HOST: a2V5Y2xvYWstcG9zdGdyZXNxbA==
>   POSTGRES_PASSWORD: ZXhhbXBsZS1rZXljbG9hay1lTWFQ
>   POSTGRES_SUPERUSER: dHJ1ZQ==
>   POSTGRES_USERNAME: ZXhhbXBsZS1rZXljbG9hay1WbGdU
> kind: Secret
> metadata:
>   labels:
>     app: keycloak
>   name: keycloak-db-secret
>   namespace: keycloak
> type: Opaque
> ```
> 
> The `Keycloak` Custom Resource also needs to be updated to turn the external database support on.
> Here's an example:
> 
> .`Keycloak` Custom Resource with external database support
> ```yaml
> apiVersion: keycloak.org/v1alpha1
> kind: Keycloak
> metadata:
>   labels:
>     app: sso
>   name: example-keycloak
>   namespace: keycloak
> spec:
>   externalDatabase:
>     enabled: true
>   instances: 1
> ```

```

Had to quote the copied examples due to backticks.

Reply:
I restructured the docs. Please have a look once you have a chance.

Reply:
The docs are looking much clearer now in my opinion. Just on this line:
https://github.com/keycloak/keycloak-operator/pull/65/files#diff-11ba1e9a3a3933aaafa03e5e34f91569R32

Since this is the only time the user will be creating the database secret, I don't believe they will be familiar with the existing properties required. Would you agree or am I missing something here?

Reply:
Fair point. I added proper docs around it.